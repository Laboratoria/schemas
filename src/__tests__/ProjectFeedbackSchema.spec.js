const mongoose = require('mongoose/browser');
const {
  CohortSchema,
  ProjectFeedbackSchema,
  ProjectSchema,
  ReviewerSurveySchema,
} = require('../../')(mongoose);

describe('ProjectFeedbackSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ProjectFeedbackSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate example', (done) => {
    const project = new mongoose.Document({}, ProjectSchema);
    const cohort = new mongoose.Document({}, CohortSchema);
    const reviewerSurvey = new mongoose.Document({}, ReviewerSurveySchema);

    const projectFeedback = new mongoose.Document({
      project: project._id,
      cohort: cohort._id,
      uid: '9x7YelqRH8hX3QRz0qV6IAhYlek1',
      createdBy: '<UID>',
      createdAt: new Date(),
      rubric: '2',
      rubricResults: {
        logic: 5,
        architecture: 3,
        communication: 4,
        github: 5,
      },
      reviewerSurvey: reviewerSurvey._id,
      reviewerSurveyResults: {
        perception: 2,
        soft: 'soft comment',
        dropout: 3,
        tech: 'tech comment',
        engagement: 1,
      },
      notes: 'revisar esto:\n-\n-\n-',
    }, ProjectFeedbackSchema);

    projectFeedback.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
