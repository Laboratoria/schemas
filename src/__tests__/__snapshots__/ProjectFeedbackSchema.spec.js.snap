// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFeedbackSchema should fail validation when fields missing 1`] = `
Object {
  "cohort": [ValidatorError: Path \`cohort\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "project": [ValidatorError: Path \`project\` is required.],
  "reviewerSurvey": [ValidatorError: Path \`reviewerSurvey\` is required.],
  "rubric": [ValidatorError: Path \`rubric\` is required.],
  "rubricResults": [ValidatorError: Path \`rubricResults\` is required.],
  "uid": [ValidatorError: Path \`uid\` is required.],
}
`;
